mp/importer/tests/test_batcher.py

```diff
@@ -0,0 +1,114 @@
+import os
+from unittest import TestCase
+import collections
+from collections import namedtuple
+from collections import Iterable
+from itertools import *
+
+def make_test_context(b_size, b_start, b_max):									# creates a "provisional" context 
+    context = namedtuple('Context', """
+        batch_size
+        start_batch
+        max_batches """)
+    context.batch_size = b_size
+    context.start_batch = b_start
+    context.max_batches = b_max
+    return context
+
+class TestBatcher(TestCase):
+    def test_batches(self):
+        print ("*** test make batches")
+        log = []
+        def end_batch():				
+            log.append('X')
+        from .. import batcher
+        context = make_test_context(2,0,2)
+        batcher = batcher.run_in_batches(		
+                context,
+                iter('abcdefghilm'), end_batch)			
+        for letter in batcher:				
+            log.append(letter)
+        self.assertEqual(
+                ''.join(log),
+                'abXcdX')
+
+    def test_Max_batches(self):
+        print ("*** test Max batches")
```

Comment:
These tests are repeating too much code. We can probably refactor and have less boilerplate code by moving the repeated code into a helper function.